screens/PathScreen.tsx

```diff
@@ -0,0 +1,546 @@
+/* eslint-disable react-hooks/exhaustive-deps */
```

Comment:
This component is too large. It handles navigation, data fetching, UI rendering, auto-scroll logic, and state management all in one file. Consider breaking it down into smaller, focused components:
> - PathReader - for the main reading interface
> - PathControls - for play/pause/auto-scroll controls
> - PathNavigation - for ang navigation
> - PathProgress - for progress tracking